### PR TITLE
staticanalysis/bot: Forward exceptions encountered during analyzer run(...) to Sentry.

### PR DESCRIPTION
**Pull request by abpostelnicu**
The scope of this patch is to have a method to continue the analysis process
even if an analyzer has failed.
In a case of failure, the exception is logged and pushed to Sentry.

### CODE REVIEW

File: src/staticanalysis/bot/static_analysis_bot/workflow.py [hunk 1]
```diff
@@ -279,28 +279,33 @@ def detect_issues(self, analyzers, revision, is_before=False):
...
+            except AnalysisException as ex:
+                # Log the error to Sentry
+                logger.error('Analyzer {} thrown exceptions during runtime.'.format(analyzer_class.__name__))
+                logger.error(ex.message)
+            finally:
+                # Clean up any uncommitted changes left behind by this analyzer.
+                self.hg.revert(settings.repo_dir.encode('utf-8'), all=True)
+                logger.info('Reverted all uncommitted changes in repo', rev=self.hg.identify())
+
+                # Compute line hashes now before anything else changes the code.
```
Comment (marco-c) [inline src/staticanalysis/bot/static_analysis_bot/workflow.py]: I'd move this part in the try block, so you don't need to define `analyzer_issues` before as an empty list

Comment (abpostelnicu) [reply src/staticanalysis/bot/static_analysis_bot/workflow.py]: Sure good point.

Comment (marco-c) [reply src/staticanalysis/bot/static_analysis_bot/workflow.py]: Oh wait, maybe we have to compute the line hashes *after* the revert? @La0 do you remember?

Comment (La0) [reply src/staticanalysis/bot/static_analysis_bot/workflow.py]: [Some work was done](https://github.com/mozilla/release-services/pull/1745) last december around this.

I'm not sure computing after the revert changes anything

Comment (marco-c) [reply src/staticanalysis/bot/static_analysis_bot/workflow.py]: Actually, in that thread of discussion, @jankeromnes fixed an issue by making the cleanup be done before the hash calculation: https://github.com/mozilla/release-services/pull/1745#issuecomment-447839046.

So maybe we have to keep the same order here.

**Pull request merged**